profiler/profile.go

```diff
@@ -44,48 +44,125 @@ const (
 	MetricsProfile
 )
 
-func (t ProfileType) String() string {
-	switch t {
-	case HeapProfile:
-		return "heap"
-	case CPUProfile:
-		return "cpu"
-	case MutexProfile:
-		return "mutex"
-	case BlockProfile:
-		return "block"
-	case GoroutineProfile:
-		return "goroutine"
-	case expGoroutineWaitProfile:
-		return "goroutinewait"
-	case MetricsProfile:
-		return "metrics"
-	default:
-		return "unknown"
+// collector holds the implementation details of a ProfileType, see collectors
+// map below.
+type collector struct {
+	// Name is used to lookup the profile for pprof.Lookup(name) and returned by
```

Comment:
Generally we've been trying to stick with the word "specifies" to avoid documenting how this is used and instead document what _it is_.
```suggestion
	// Name specifies the profile name as used with pprof.Lookup(name) and returned by
```

Reply:
I don't see a semantic difference as a result of rewriting this sentence, but the new sentence sounds nice as well, so I'll apply the diff.